usautobuild/actions/good_files.py

```diff
@@ -0,0 +1,120 @@
+import json
+import re
+import shutil
+import time
+import os
+
+from logging import getLogger
+from pathlib import Path
+
+import humanize
+
+from usautobuild.config import Config
+from usautobuild.exceptions import (
+    BuildFailedError,
+    InvalidProjectPathError,
+    MissingLicenseFileError
+)
+from usautobuild.utils import git_version, run_process_shell
+
+log = getLogger("usautobuild")
+
+
+class good_files:
+    def __init__(self, config):
+        self.config = config
+        self.files_to_keep_in_managed = self.get_files_to_keep_in_managed()
+
+    def get_files_to_keep_in_managed(self):
+        path = Path.cwd() / "local_repo/Tools/CodeScanning/CodeScan/CodeScan/bin/Debug/net7.0/FilesToMoveToManaged.json"
+        with open(path, 'r') as file:
+            return json.load(file)
+
+    def make_good_files_build(self, VersionTag) -> None:
+        good_files_dir = Path(self.config.output_dir) / "good_files"
+        good_files_dir.mkdir(parents=True, exist_ok=True)
+
+        # Check if bundledDLL/version.txt exists and verify the version
+        self.check_version()
+        
+        for target in self.config.target_platforms:
+            if target == "linuxserver":
+                log.debug("Skipping %s", target)
+                continue
+
+            
+            target_path = Path(self.config.output_dir) / target
+            if target_path.exists() and target_path.is_dir():
+                destination = good_files_dir / target
+                shutil.copytree(target_path, destination, dirs_exist_ok=True)
+                self.prepare_target(target, destination)
+            else:
+                log.warning("Target path %s does not exist or is not a directory", target_path)
+
+    def check_version(self):
+        version_file = Path.cwd() / "bundledDLL" / "version.txt"
+        if version_file.exists():
+            with open(version_file, 'r') as file:
+                version_info = json.load(file)
```

Comment:
why is the extensions txt if this is a json formatted file?

Reply:
from version changes, originally was just going to be x.y.z in a text file but was updated over course of stuff, Hub expects txt too

Reply:
damn, if only we had access to the source code so it expected a json file with json data

Reply:
xD haha ok ill update it